### PR TITLE
HPCC-17148 Fix various minor warnings and unreachable code

### PR DESCRIPTION
**Pull request by ghalliday**
Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [ ] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [ ] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [ ] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Comment by ghalliday** [comment]
@richardkchapman please review

File: ecl/hqlcpp/hqlecl.cpp [hunk 1]
```diff
@@ -604,11 +604,10 @@ bool HqlDllGenerator::doCompile(ICppCompiler * compiler)
     if (ok && deleteGenerated)
     {
         StringBuffer temp;
-        remove(temp.clear().append(wuname).append(".cpp").str());
-        remove(temp.clear().append(wuname).append(".hpp").str());
+        removeFileTraceIfFail(temp.clear().append(wuname).append(".hpp").str());
```
Comment (richardkchapman) [inline ecl/hqlcpp/hqlecl.cpp]: Did you mean to delete the .cpp line?

File: system/jlib/jcomp.cpp [hunk 2]
```diff
@@ -722,26 +722,26 @@ StringBuffer & CppCompiler::getObjectName(StringBuffer & out, const char * filen
...
+            removeFileTraceIfFail(temp.clear().append(targetDir).append(coreName).append(".exp").str());
+            removeFileTraceIfFail(temp.clear().append(targetDir).append(coreName).append(".lib").str());
+            removeFileTraceIfFail(temp.clear().append(targetDir).append(coreName).append(".res").str());
+            break;
+        }
     case GccCppCompiler:
         {
-            StringBuffer temp;
-            remove(temp.clear().append(targetDir).append(coreName).append(".exp").str());
-            remove(getObjectName(temp.clear(), coreName).str());
```
Comment (richardkchapman) [inline system/jlib/jcomp.cpp]: This line has been lost I think... was it deliberate?

**Comment by richardkchapman** [comment]
@ghalliday Some comments. Also you didn't fill in any information about how/if you have tested this... or did you consider the changes too minor to need testing (always a danger sign!)

**Comment by ghalliday** [comment]
Yes I should have done
- Ran the full regression test.
- Modified the remove code to lose the 1st character and check what warnings came out.
- restored it and checked what warnings came out.
From that realised that several files were being deleted more than once, and others were only generated on windows - hence the changes to what is deleted.

File: ecl/hqlcpp/hqlecl.cpp (hunk 1 above)
Comment (ghalliday) [inline ecl/hqlcpp/hqlecl.cpp]: yes - it was already deleted by code elsewhere.

File: system/jlib/jcomp.cpp (hunk 2 above)
Comment (ghalliday) [inline system/jlib/jcomp.cpp]: yes - there is no a.obj - it is a.cpp.obj.

**Comment by ghalliday** [comment]
@richardkchapman updated with details of testing and replied to comments

**Pull request merged**